admin/api/src/main/java/org/codice/ddf/admin/api/config/sources/WfsSourceConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.admin.api.config.sources;
+
+import static org.codice.ddf.admin.api.validation.SourceValidationUtils.validateWfsFactoryPid;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.codice.ddf.admin.api.config.ConfigurationType;
+import org.codice.ddf.admin.api.handler.ConfigurationMessage;
+import org.codice.ddf.admin.api.validation.ValidationUtils;
+
+import com.google.common.collect.ImmutableMap;
+
+public class WfsSourceConfiguration extends SourceConfiguration {
+
+    public static final String CONFIGURATION_TYPE = "wfs-source";
+    private static final Map<String, Function<SourceConfiguration, List<ConfigurationMessage>>> FIELDS_TO_VALIDATION_FUNC = new ImmutableMap.Builder<String, Function<SourceConfiguration, List<ConfigurationMessage>>>()
+            .putAll(getBaseFieldValidationMap())
+            .put(FACTORY_PID, config -> validateWfsFactoryPid(config.factoryPid(), FACTORY_PID))
+            .build();
+
+    public WfsSourceConfiguration() {
```

Comment:
They are being used by the ConfigurationHandlers to get the ConfigurationType, we can investigate making them static instead. Although, I'm sure these constructors will look nicer in unit tests with the builder methods.